core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java

```diff
@@ -171,7 +173,15 @@ public Set<Long> getReplicationLimitedFileIds() {
    * @return the root of the inode tree
    */
   public InodeDirectory getRoot() {
-    return mInodeStore.get(0).map(Inode::asDirectory).orElse(null);
+    if (mRoot == null) {
+      mRoot = mInodeStore.get(0).map(Inode::asDirectory).orElse(null);
+      synchronized (this) {
+        if (mRoot == null) {
```

Comment:
If another thread has updated `mRoot` and in the current thread (CPU) `mRoot` is still null, will we have true or false on this line @elega ? Would it be better to `synchronized(mRoot)`?